src/Symfony/Component/ObjectMapper/ArrayMapper.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper;
+
+use Symfony\Component\ObjectMapper\Exception\MappingException;
+use Symfony\Component\ObjectMapper\Exception\WrappedMappingException;
+
+/**
+ * Map an array of objects into the specified target array using the ObjectMapper.
+ *
+ * @experimental
+ *
+ * @author Martin Komischke <martin.komischke@gmail.com>
+ */
+final class ArrayMapper
+{
+    public function __construct(
+        private ObjectMapperInterface $mapper,
+        private string $throwPolicy = CollectionMapperThrowPolicy::FAIL_SAFE,
+    ) {
+    }
+
+    public function map(array $source, array $target): void
+    {
+        \assert(\count($source) === \count($target));
```

Comment:
`assert` should not be used in that context as it validates developer inputs. And looking at the code below, the keys must match.

```suggestion
        if (array_keys($source) !== array_keys($target)) {
            throw new \InvalidArgumentException('Source and target array must have the same keys');
        }
```